src/custom/pages/Claim/InvestmentFlow/InvestOption.tsx

```diff
@@ -58,16 +58,16 @@ export default function InvestOption({ approveData, claim, optionIndex }: Invest
 
     updateInvestAmount({ index: optionIndex, amount })
     setTypedValue(value.toExact() || '')
-    setInputError('')
+    updateInvestError({ index: optionIndex, error: undefined })
 
     setPercentage(_formatPercentage(calculatePercentage(balance, maxCost)))
-  }, [balance, maxCost, noBalance, optionIndex, updateInvestAmount])
+  }, [balance, maxCost, noBalance, optionIndex, updateInvestAmount, updateInvestError])
 
   // on input field change handler
   const onInputChange = useCallback(
     (value: string) => {
       setTypedValue(value)
-      setInputError('')
+      updateInvestError({ index: optionIndex, error: undefined })
```

Comment:
nit: make a handleResetError or similar to clean these up and make them more declarative.

Reply:
Updated